Add image dependencies for nRF53 upgrade candidates (#7670)
nrfconnect/sdk-nrf: doc/nrf/releases/release-notes-changelog.rst

      * The image generation script by introducing the sub-element structure inside the Zigbee OTA image.
        Enable :kconfig:option:`CONFIG_ZIGBEE_FOTA_GENERATE_LEGACY_IMAGE_TYPE` to generate images compatible with previous NCS releases.
      * Default value of the :kconfig:option:`CONFIG_ZIGBEE_FOTA_IMAGE_TYPE` to 0x0141.

>       * Enabled the :kconfig:option:`CONFIG_NRF53_ENFORCE_IMAGE_VERSION_EQUALITY` for nRF5340 SoC to ensure integrity of the upgrade image.

Reviewer: ```suggestion
      * Setting of the :kconfig:option:`CONFIG_NRF53_ENFORCE_IMAGE_VERSION_EQUALITY` for nRF5340 SoC to ensure integrity of the upgrade image.
        Now this Kconfig option is enabled by default.
```
Let's match the style.

The following list summarizes both the main changes inherited from upstream MCUboot and the main changes applied to the |NCS| specific additions:

* Added :kconfig:option:`CONFIG_NRF53_ENFORCE_IMAGE_VERSION_EQUALITY` to attach the dependencies between application and network core images.

>   This option will link the upgrade images in a way, that either both or none will be applied.

Reviewer: ```suggestion
  This option links the upgrade images in such a way that either both or none is applied.
```
